src/plugins/commander.go

```diff
@@ -163,11 +163,14 @@ func (c *Commander) agentRegistered(cmd *proto.Command) {
 				}
 			}
 
-			for index, feature := range agtCfg.Details.Features {
-				agtCfg.Details.Features[index] = strings.Replace(feature, "features_", "", 1)
+			if agtCfg.Details != nil && agtCfg.Details.Features != nil {
+				for index, feature := range agtCfg.Details.Features {
+					agtCfg.Details.Features[index] = strings.Replace(feature, "features_", "", 1)
+				}
+
+				sort.Strings(agtCfg.Details.Features)
 			}
 
-			sort.Strings(agtCfg.Details.Features)
 			sort.Strings(c.config.Features)
```

Comment:
Need feedback on `sort.Strings(c.config.Features)`.

I have not added nil check around it with assumption `c.config.Features` won't be null.
Is my assumption true?

Reply:
I think we're OK on this line,`sort.Strings` works fine if given a `nil`: https://go.dev/play/p/91Jt0vRD9vv

The next few lines look like more potential nil pointer derefs if `agtCfg.Details` is nil. Maybe we make one larger `if agtCfg.Details != nil {` block and pull everything down to L189 into it? Seems like this code is all about synchronizing `agtCfg.Details.Features` and `c.config.Features`, and we don't need to do any of that work if we don't have any `agtCfg.Details`.

It's a little tempting to introduce a helper methods here, so we could have something simple in `agentRegistered`:

```go
if agtCfg.Details != nil && agtCfg.Details.Features != nil {
  c.synchronizeFeatures(agtCfg.Details.Features)
}
```

Reply:
At first I thought the same, but agent uses `agentConfig` as a way to see if management and data plane are sync. It's likely if `agtCfg.Details.Features` is nil but `c.config.Features` not. Agent would flag such setting as un-synchronized. There fore take steps to synchronize.

If we go by above approach, synchronization would then only happen when `agtCfg.Details.Features` is non-nil.

cc @oliveromahony

Reply:
hrm, just noticed that L141 kinda looks like it's doing a lot of the same work?

Reply:
Oh yeah, it's sth that is taken care of before this. I will refactor this a bit. Thanks for pointing this.